test/expr.jl

```diff
@@ -154,6 +154,13 @@ function expressions_test(
         @test k == 0
     end
 
+    @testset "getindex(aff::AffExpr, v::VariableRef)" begin
+        m = ModelType()
+        x = @variable(m, x)
+        aff = @expression(m, 1.0 * x)
+        @test aff[x] == 1.0
+    end
```

Comment:
Some other tests we need 
- Affine
  - Checking that a missing term returns `0.0`
  - Checking that the quadratic term returns `0.0`
- Quadratic
  - `quad[x, y]` and `quad[y, x]` return the same
  - `quad[x]` returns the affine component
  - missing values return 0
- Variable
  - `x[x]` returns 1
  - `x[y]` returns 0
  - `x[x, y]` returns 0

Reply:
Adding to this - if a variable does not exist in a model but you check an expression for its coefficient, should this throw an error?

Reply:
You could use https://github.com/jump-dev/JuMP.jl/blob/ba17068a2f689e582604b8f030238fd19f5d8971/src/variables.jl#L155
but it's probably not important. It will return `0` anyway.